core/domain/exp_jobs_one_off.py

```diff
@@ -535,3 +539,48 @@ def reduce(key, values):
         # Combine all values from multiple lists into a single list
         # for that error type.
         yield (key, list(set().union(*final_values)))
+
+
+class ImageDataMigrationJob(jobs.BaseMapReduceOneOffJobManager):
+    """One-off job for migrating the images in the exploration
+    from the GAE to GCS.
+    """
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [file_models.FileSnapshotContentModel]
+
+    @staticmethod
+    def map(file_snapshot_content_model):
+        instance_id = (
+            file_snapshot_content_model.get_unversioned_instance_id())
+        filetype = instance_id[instance_id.rfind('.') + 1:]
+        # to filter all the image entries.
+        if filetype in ALLOWED_IMAGE_EXTENSIONS:
+            pattern = re.compile(
+                r"^/([^/]+)/assets/([^/]+)\.(png|jpg|gif|jpeg)$")
+            parts_of_id = pattern.match(instance_id)
+            if not parts_of_id:
+                yield('Error: The instance_id is not correct', instance_id)
+            else:
+                filename = parts_of_id.group(2) + '.' + parts_of_id.group(3)
+                filepath = 'assets/' + filename
+                exploration_id = parts_of_id.group(1)
+                file_model = file_models.FileModel.get_model(
+                    exploration_id, filepath, False)
+                if file_model:
+                    if file_model.deleted:
+                        yield('Error: found deleted file', file_model.id)
+                    else:
+                        # content = file_model.content
+                        # fs = fs_domain.AbstractFileSystem(
+                        #   fs_domain.GcsFileSystem(exploration_id))
+                        # fs.commit(file_model.user_id, '%s/%s' %('images',
+                        #   filename, content)
+                        yield('File Copied', 1)
```

Comment:
Pull the string into a named constant and use this constant everywhere. Otherwise it's less maintainable -- if we change the chars in one of the string then we might miss doing so in the other places.

Reply:
Done1